library/src/main/java/com/getstream/sdk/chat/viewmodel/CreateChannelViewModel.kt

```diff
@@ -30,31 +35,31 @@ class CreateChannelViewModel(
     }
 
     private fun queryChannel(channelName: String) {
-        val channelId: String = channelName.replace(" ".toRegex(), "-").toLowerCase()
         val members = client.getCurrentUser()?.run {
             listOf(Member(this))
         } ?: listOf()
         val channel = Channel().apply {
-            this.cid = "messaging:$channelId"
-            this.id = channelId
+            this.cid = "messaging:$channelName"
+            this.id = channelName
             this.type = "messaging"
             this.name = channelName
             this.members = members
         }
-        domain.useCases.createChannel.invoke(channel).execute().run {
+        viewModelScope.launch(ioDispatcher) {
+            val result = domain.useCases.createChannel.invoke(channel).execute()
             when {
-                isSuccess -> {
+                result.isSuccess -> {
                     stateMerger.postValue(State.ChannelCreated)
                 }
-                isError -> {
+                result.isError -> {
                     stateMerger.postValue(State.BackendError)
                 }
             }
         }
     }
 
     private fun validateChannelName(channelNameCandidate: String): Boolean {
-        return channelNameCandidate.isNotEmpty()
+        return channelNameCandidate.isNotEmpty() && channelNameCandidate.matches(Regex("^!?[\\w-]*\$"))
```

Comment:
Using a RegEx should be enought, `isNotEmpty()` call is redundant.
On the other hand, could we extract RegEx into a constant that describes what this RegEx is used for?

Reply:
This regexp accepts an empty string in Kotlin, so I'll leave isNotEmpty() check. 
I'll extract regex to give it a name.